### DISCUSSION TITLE
Can we run a full-vehicle crash model made for LS Dyna, (.key) input file with open radioss?

### DISCUSSION THREAD

**Discussion by Popular1319** [OpenRadioss/OpenRadioss#1729] [Users]
I've tried running a model with a node count of 2 million and I've received an error as shown below, can anyone help me debug the cause?
I'm attaching an image of the error that I've got.![tmp_5208385f-dc5e-4bd6-95e9-83bb760b6ff8.png](https://github.com/OpenRadioss/OpenRadioss/assets/140241644/a251972a-5543-4f9e-8b19-3eae71e81f42)

**Comment by PaulAltair** [comment]
It looks from your screenshot, like you are trying to use a runscript that is expecting a Radioss file as input, you can submit the .key to starter from command line/terminal directly (not using that script), modify the script, or use the python/tk guis (see discussion in this forum: https://github.com/orgs/OpenRadioss/discussions/1231#top) or the confluence page at [Confluence page for job submission guis](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609/python+tk+guis+for+job+submission+anim-vtk+conversion+and+T01-csv+conversion) and it should 'work', whether the job actually runs beyond that will depend on the specific Dyna content. There are some examples of working models (CCSA Yaris and Camry) on the models page at OpenRadioss.org https://www.openradioss.org/models/

**Reply by Popular1319** [reply]
Thankyou @PaulAltair. I'll try using the .key format gui to run.

**Reply by Popular1319** [reply]
Hi @PaulAltair ,

I've tried using this script file (attached below) after that also I'm trying to run using 64 threads and 32 cores. The Camry model initiated run successfully using the same parameters. 
I'm attaching the command line outputs in
 msg_file_2.txt![msg_file_2.txt](https://github.com/OpenRadioss/OpenRadioss/files/12144372/msg_file_2.txt)


![tmp_9b4f43c7-6781-4403-981b-e9e14e186d6e.png](https://github.com/OpenRadioss/OpenRadioss/assets/140241644/8e387e20-be60-4710-9f5e-0c89cd1bdbd0)


The run_mpi.bat is :-

```
@ECHO OFF
call "C:\Program Files (x86)\Intel\oneAPI\mpi\2021.9.0\env\vars.bat"
set /p input= put input name (without.key) :
set /p np= put  number of np :
set OPENRADIOSS_PATH=F:\02_open_radioss\OpenRadioss
echo %OPENRADIOSS_PATH%
set RAD_CFG_PATH=%OPENRADIOSS_PATH%\hm_cfg_files
echo %RAD_CFG_PATH%
set OMP_SET_NUM_THREADS=64
set RAD_H3D_PATH=%OPENRADIOSS_PATH%\extlib\h3d\lib\win64
echo %RAD_H3D_PATH%
set KMP_STACKSIZE=999m
#set PATH=%OPENRADIOSS_PATH%\extlib\hm_reader\win64;%PATH%
set PATH=%OPENRADIOSS_PATH%\extlib\hm_reader\win64;%PATH%;%OPENRADIOSS_PATH%\extlib\intelOneAPI_runtime\win64;%PATH%;"C:\Program Files (x86)\Intel\oneAPI\mpi\2021.9.0\bin"
set MPI_PATH="C:\Program Files (x86)\Intel\oneAPI\mpi\2021.9.0\bin"
echo %PATH%
echo "**Run Starter**"
#echo %myjobname% "running in OpenRadioss"
%OPENRADIOSS_PATH%\exec\starter_win64.exe -i %input%.key -np %np% >starter.out 
echo "**Run Engine**"
%MPI_PATH%\mpiexec.exe -delegate -np %np% %OPENRADIOSS_PATH%\exec\engine_win64_impi.exe -i %input%_0001.rad  >engine.out
echo "Done"

```
I'm unable to find the cause of this error, in starter.out also nothing is printed because of sudden termination.

**Reply by PaulAltair** [reply]
I don't know then, can you share the file you are trying to run, it may be an issue with the input, rather than the submission line if the Camry is ok?

Have you tried it on a lower core count? Does it work e.g. for -nt 2, -np 2?

I have a couple of comments too:

OMP_SET_NUM_THREADS doesn't work for me, (I have to set -nt in the Starter submission)

and in any case 64 is way more threads than you would normally use, I wouldn't normally go over 4-8

Do you actually have 2048 cpu to use?

**Reply by elequiniou** [reply]
Hi, my understanding is that you would like to use 64 processes on a 32 cores machine, using HyperThreading (HT), correct ?
If so, as you already set -np 32, then "set OMP_SET_NUM_THREADS=2" and not 64 to use 32x2=64 threads in total.
Take care that the benefit from HT will be marginal, as OpenRadioss already saturates the memory bandwidth running 32 MPI on 32 physical cores. Hope this help to fix the error (certainly due to too many processes generated by your 32 MPI x 64 threads = 2048 threads job in total as accounted by Paul.
Thanks a lot lot!
Eric

**Reply by Popular1319** [reply]
> I don't know then, can you share the file you are trying to run, it may be an issue with the input, rather than the submission line if the Camry is ok?
> 
> Have you tried it on a lower core count? Does it work e.g. for -nt 2, -np 2?
> 
> I have a couple of comments too:
> 
> OMP_SET_NUM_THREADS doesn't work for me, (I have to set -nt in the Starter submission)
> 
> and in any case 64 is way more threads than you would normally use, I wouldn't normally go over 4-8
> 
> Do you actually have 2048 cpu to use?

Hi @PaulAltair I was unsure of -nt count and thought it as processes so, I'll try changing with new -nt count as 2 or 4.

**Reply by Popular1319** [reply]
> Hi, my understanding is that you would like to use 64 processes on a 32 cores machine, using HyperThreading (HT), correct ?
> If so, as you already set -np 32, then "set OMP_SET_NUM_THREADS=2" and not 64 to use 32x2=64 threads in total.
> Take care that the benefit from HT will be marginal, as OpenRadioss already saturates the memory bandwidth running 32 MPI on 32 physical cores. Hope this help to fix the error (certainly due to too many processes generated by your 32 MPI x 64 threads = 2048 threads job in total as accounted by Paul.
> Thanks a lot lot!
> Eric

Thank you @elequiniou now I got a proper understanding of the threads and cores. The model which I'm running is a proprietary model I'm not sure I can share that here.

**Reply by PaulAltair** [reply]
Have you had any more success? Still not sure if this an exe issue or model issue. If you can't share the model, then you can check the input for the used DYNA cards vs those supported? Though it is odd that here the starter does not even seem to begin!

**Reply by Popular1319** [reply]
Hi @PaulAltair I've installed the new gui job submission tool which also takes .k and .key as input. I've prepared a sample model shows the as shown below
![image](https://github.com/OpenRadioss/OpenRadioss/assets/140241644/446897c8-18b5-48a2-a971-8003b987eea4)
Also I couldn't see any error in the command window. Can you give me some reasons why this error happens?

**Reply by Popular1319** [reply]
Also I've checked that my model in which there are no unsupported cards and is error free.

**Reply by PaulAltair** [reply]
If you can share your sample model I can have a look?

Have you tried the simple tensile test dyna model on the models page to test if your install/setup is correct? 

[Tensile Test in DYNA format on confluence model page](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/21364914/Tensile+Test+Model+in+LS-DYNA+format)
